rskj-core/src/main/java/co/rsk/trie/NodeReference.java

```diff
@@ -63,7 +73,14 @@ public boolean isEmpty() {
 
         Optional<Trie> node = store.retrieve(lazyHash.getBytes());
 
-        lazyNode = node.orElse(null);
+        // Broken database, can't continue
+        if (!node.isPresent()) {
+            nodeStopper.stop(1);
+            return Optional.empty();
```

Comment:
A really clear and to the point log on _why_ your node is stopping should be present, because if you are here your node won't start again (or rather, will exist always at the same point). Can be very confusing for an unsuspecting user.

Reply:
Done!